### PR TITLE
Added File IO Testing

### PR DESCRIPTION
**Pull request by auroq**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][1]! For simplicity, please make sure that your pull request includes one and only one contribution.

## Complete the Applicable Sections Below

Find which section best describes your pull request and make sure you fill it out. To start, let us know which issue you've fixed.

- [ ] I fixed #your-issue-number-here

### Code Snippets

- [ ] I named the pull request using `Added/Updated <Sample Program> in <Language>` format
- [ ] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])

### Documentation

- [ ] I named the pull request using `Added/Updated <Sample Program> in <Language> Article` format
- [ ] I followed the applicable article template
  - [ ] I added/updated a [language article][4] (i.e. The Python Programming Language)
  - [ ] I added/updated a [code snippet article][3] (i.e. Hello World in Perl)
  - [ ] I added/updated a [project article][5] (i.e. Fizz Buzz in Every Language)

### Testing

- [x]  I names the pull request using `Added/Updated <Language>/<Project> Testing` format
- [ ] I followed the testinfo template, if applicable

## Notes

Feel free to describe what you added or updated.

[1]: https://therenegadecoder.com/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[3]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/CODE_ARTICLE_TEMPLATE.md
[4]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/LANGUAGE_ARTICLE_TEMPLATE.md
[5]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/PROJECT_ARTICLE_TEMPLATE.md

### CODE REVIEW

**Comment by auroq** [comment]
In order to test File IO, I needed access to the output.txt file that is created at runtime, meaning that I needed access to the container itself.
In order to do that I added an exec method to `Source`, but I'm worried that could be abused. If we want to preven abuse, I can get rid of the `exec()` method and replace it with a more specific `get_file_text()` method.

File: test/projects/test_file_io.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+@pytest.fixture(params=project_permutations[ProjectType.FileIO].params,
+                ids=project_permutations[ProjectType.FileIO].ids,
+                scope='module')
+def file_io(request):
+    request.param.build()
+    yield request.param
+    request.param.cleanup()
+
+
+def test_baklava(file_io):
```
Comment (jrg94) [inline test/projects/test_file_io.py]: Is this supposed to say "test_baklava"?

**Comment by jrg94** [comment]
Random idea: do you think it makes sense to pull your testing code out into its own repo? I was thinking it might be beneficial to start versioning your work and maybe even pip package it.

File: test/projects/test_file_io.py (hunk 1 above)
Comment (auroq) [reply test/projects/test_file_io.py]: Good catch

**Comment by auroq** [comment]
> **Random idea**: do you think it makes sense to pull your testing code out into its own repo? I was thinking it might be beneficial to start versioning your work and maybe even pip package it.
> 
> I can make a separate repo under The Renegade Coder (or you can use your own if you'd like).

That is something I have been considering as well.

Since it was originally pushed here, it is property of TheRenegadeCoder. However, with your consent, I would love to take ownership of it under my own name (but using the MIT license).

**Comment by jrg94** [comment]
Yeah, go for it! It's all your work after all. That way you can give it a cool name and whatnot. 👍

**Comment by jrg94** [comment]
I actually just used this [GitHub tutorial](https://help.github.com/en/articles/splitting-a-subfolder-out-into-a-new-repository) today to create a new repo with all the history maintained.

**Pull request merged**